### DISCUSSION TITLE
Longer measurements and raspberry pi 4 + ADC

### DISCUSSION THREAD

**Discussion by Klipton1** [fhdm-dev/scoppy#74] [Q&A] [answered]
Hello all,

I am new to posting on github so sorry if I am breaking a rule.

Is there a way to measure more samples then 2048? This could be nice when doing data acquisition for FRF measurements. As more data results in a more accurate FRF.

Furthermore this works with the raspberry pi Pico now. Are there ideas to bring this to the raspberry pi 4 + high accurate ADC? This could result in a cheap and accurate oscciloscope platform.

Kind regards,
Klipton

**Comment by fhdm-dev** [comment, marked as answer]
Hi and welcome to the Scoppy forum,

The only way to get more samples is by doing a single shot capture. You'll get 100k samples (shared between channels).

Having said that, adding the ability to specify the number of samples is on my to-do list :smile: 

There's no plans to support the RPi4 because I think the Pico/RP2040 would work fine with an external ADC (depending on the speed/accuracy you're after of course). I've done some work in this area but it's still in the early stages of development.

**Reply by Klipton1** [reply]
Thanks for the reply!

I will try doing the single shot capture. I really like the project and also recommended it to my old teachers and fellow hobbyists.

Kind regards,
Klipton1
